src/Compilers/CSharp/Portable/Lowering/SynthesizedMethodBaseSymbol.cs

```diff
@@ -106,7 +106,7 @@ public sealed override ImmutableArray<ParameterSymbol> Parameters
             {
                 if (_parameters.IsDefault)
                 {
-                    ImmutableInterlocked.InterlockedCompareExchange(ref _parameters, MakeParameters(), default(ImmutableArray<ParameterSymbol>));
+                    ImmutableInterlocked.InterlockedInitialize(ref _parameters, MakeParameters());
```

Comment:
Can you help me understand how these two lines are different? I'm missing the distinction here.

Reply:
➡️ They are semantically identical. The simpler form allows a reader to understand the behavior without the mental burden of relating the third parameter of `InterlockedCompareExchange` to the default state for the field.